lib/index.js

```diff
@@ -1097,6 +1101,11 @@ vmadm.stop = function vmStop(opts, callback) {
         args.push('-F');
     }
 
+    if (opts.timeout) {
+        // all args are required to be strings
+        args.push('-t', opts.timeout.toString());
+    }
+
```

Comment:
> Without understanding the broader context, it's hard to say whether its
> worth it. From an abstract point of view, the fact that I can ask for
> something and have no way of knowing if it was valid or not, seems
> unfortunate.

It's unfortunate, but at this point adding validation to vmadm will likely be a major version upgrade since a new vmadm that validates its parameters will break anyone who is currently using parameters that are ignored. There's a ticket open (OS-2055) for adding validation but there is no schedule set for that work.

> What are the guarantees that we're trying to make? Given the way that
> vmadm seems designed here, it definitely limits our options as it cannot
> provide that feedback.

In the context of *this* change: node-vmadm is used only by the SDC agents currently, and the agents are required to work on all versions of SmartOS currently deployed with SDC. This means we cannot assume that there's any validation in vmadm (since no version has had it yet) nor notification from the system that -t does or hasn't worked if we do try to use it. Since the platform also doesn't include any mechanism for us to detect features (OS-4121) there's no way for an agent to know whether the platform has support for -t or not. So actually in order to correctly notice this will do nothing here we'd actually need to:

 * include a minimum version for -t support (the platform where OS-5208 goes in) in node-vmadm
 * have node-vmadm check the platform version for stop actions against the minimum version
 * do an additional VM load operation before each stop action that includes a timeout (which will be all docker stop actions I believe, since sdc-docker includes a default too)

After doing that, we could then add a warning to the log that the timeout was being ignored if either the current platform is too old to support this feature or if the timeout is being passed and the VM in question is not a docker container. The other alternative would be to fail the stop job in this latter case. That seems a bit extreme given that again the agents have historically ignored all unknown parameters.

Since the best we can do is log a warning that the -t is being ignored, it doesn't seem to me that it's worth slowing down all docker stop jobs for.